### ISSUE TITLE
Support for Android 9.0 Pie

### ISSUE THREAD

**Issue by Lanchon** [Lanchon/haystack#23]
support for Pie is blocked by a javac bug that i describe [here](https://gist.github.com/Lanchon/383f05664d59b096dc9d58ba32cdd667). i reported the bug to bugreport.java.com and it has been assigned internal review ID 9059540. the bug is present in both open-jdk 8 and 9.

**UPDATE:** bug has been accepted [here](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8219810).

(FYI: [this report](https://stackoverflow.com/questions/33302349/java-compiler-fails-when-compiling-simple-program) could be related.)

as a workaround you can apply the hook and core patches for android 8 on android 9 and they will provide always-on signature spoofing. but the optional UI patch for Oreo does not apply on Pie.

**Comment by Lanchon** [comment]
i worked around the javac bug and pushed core support for android 9. UI support is not implemented for now.

**Comment by roobre** [comment]
Hey,

I just tried core patches on LOS 16 and I get:

```>>> target directory: hime-los-16__sigspoof-core
>>> apply patch: services.jar
>>> dexpatcher --api-level 28 --verbose --output hime-los-16__sigspoof-core/tmp/services.jar/patched-dex --multi-dex hime-los-16/services.jar patches/sigspoof-core/services.jar.dex
DexPatcher version 1.6.2 by Lanchon (https://dexpatcher.github.io/)
info: read 'hime-los-16/services.jar'
info: read 'patches/sigspoof-core/services.jar.dex'
error: type 'com.android.server.pm.GeneratePackageInfoHook': (GeneratePackageInfoHook.java:0): target not found
1 error(s), 0 warning(s)
```

Tested just now @ 7acdca6cd9539f37bc604d24b6b3cc97ca1ca4d8. Please tell me if you need more debug info.

**Comment by Lanchon** [comment]
did you apply the hook patch before the core patch?

**Comment by roobre** [comment]
I did... but I did not apply the core patch over the `__sigspoof-hook` directory. Definitely my bad for misreading the README.

Both the patcher and the patch are working fine now, thanks!

**Comment by Lanchon** [comment]
glad it worked, thanks! ill delete your messages because they dont belong here. next time you have an issue, please open a new issue (unless yours is already reported). thanks!

**Comment by Lanchon** [comment]
BTW, i pushed UI support for android 9; haven't made a proper release yet.

**Comment by Lanchon** [comment]
all done.
